plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt

```diff
@@ -707,4 +749,136 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
         codeModernizerBottomWindowPanelManager.reset()
         isModernizationInProgress.set(false)
     }
+
+    suspend fun getArtifactForHil(): CodeTransformHilDownloadArtifact? {
+        if (codeTransformationSession?.getHilDownloadArtifact() != null) {
+            return codeTransformationSession?.getHilDownloadArtifact()
+        }
+
+        val jobId = codeTransformationSession?.getActiveJobId() as JobId
+        val downloadArtifactId = codeTransformationSession?.getHilDownloadArtifactId() as String
+
+        val tmpDir = try {
+            createTempDirectory("", null)
+        } catch (e: Exception) {
+            val errorMessage = "Unexpected error when creating tmp dir for HIL: ${e.localizedMessage}"
+            LOG.error { errorMessage }
+            throw e
+        }
+        try {
+            val hilArtifact = artifactHandler.downloadHilArtifact(jobId, downloadArtifactId, tmpDir)
+            if (hilArtifact != null) {
+                codeTransformationSession?.setHilTempDirectoryPath(tmpDir.toPath())
+                codeTransformationSession?.setHilDownloadArtifact(hilArtifact)
+            }
+            return hilArtifact
+        } catch (e: Exception) {
+            return null
+        }
+    }
+
+    fun createDependencyReport(hilDownloadArtifact: CodeTransformHilDownloadArtifact): MavenDependencyReportCommandsResult {
+        val tmpDirPath = codeTransformationSession?.getHilTempDirectoryPath() as Path
+
+        try {
+            val copyPomForDependencyReport = createFileCopy(hilDownloadArtifact.pomFile, getPathToHilDependencyReportDir(tmpDirPath).resolve(HIL_POM_FILE_NAME))
+            setDependencyVersionInPom(copyPomForDependencyReport, hilDownloadArtifact.manifest.sourcePomVersion)
+        } catch (e: Exception) {
+            val errorMessage = "Unexpected error when preparing for HIL dependency report: ${e.localizedMessage}"
+            telemetry.error(errorMessage)
+            LOG.error { errorMessage }
+            return MavenDependencyReportCommandsResult.Failure
+        }
+
+        return codeTransformationSession?.createHilDependencyReportUsingMaven() as MavenDependencyReportCommandsResult
+    }
+
+    fun findAvailableVersionForDependency(pomGroupId: String, pomArtifactId: String): Dependency? {
+        try {
+            val report = parseXmlDependenciesReport(
+                getPathToHilDependencyReport(codeTransformationSession?.getHilTempDirectoryPath() as Path)
+            )
+            return report.dependencies?.first {
```

Comment:
Blocker: We should have some early exit here.

```
if (report.status === "no available versions") {
   // early exist and call resume transformation
}
```

Will need to update parseXmlDependenciesReport function. See other comment

Reply:
<img width="1064" alt="Screenshot 2024-05-07 at 1 56 48 PM" src="https://github.com/aws/aws-toolkit-jetbrains/assets/38665452/e208421b-dfb9-400c-9b33-803a11f6f9c4">


Here is vscode implementation

Reply:
No alternative versions found should already trigger clean up and exit early.